src/ui/public/stringify/__tests__/_string.js

```diff
@@ -17,4 +17,12 @@ describe('String Format', function () {
     expect(string.convert('Zm9vYmFy')).to.be('foobar');
   });
 
+  it('convert a string to title case', function () {
+    var StringFormat = fieldFormats.getType('string');
+    var string = new StringFormat({
+      transform: 'title'
+    });
+    expect(string.convert('PLEASE DO NOT SHOUT')).to.be('Please Do Not Shout');
```

Comment:
Can you add tests that explain the other functionality of the regex, such as the fact that it maintains punctuation. I don't want someone to see this code later and say "Oh, there's a lodash function that does this!".